tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -898,6 +909,98 @@ def fit_extra(self, hyperparameters, time_limit=None, base_model_names=None, **k
+            num_rows = len(y_pred_proba_max)
+
+            if curr_percentage > max_percentage or curr_percentage < min_percentage:
+                if curr_percentage > max_percentage:
+                    num_rows_threshold = max(np.ceil(max_percentage * num_rows), 1)
+                else:
+                    num_rows_threshold = max(np.ceil(min_percentage * num_rows), 1)
+                y_pred_proba_max_sorted = y_pred_proba_max.sort_values(ascending=False, ignore_index=True)
+                # Set current threshold to num_rows_threshold - 1
+                curr_threshold = y_pred_proba_max_sorted[num_rows_threshold - 1]
+
+            # Pseudo indices greater than threshold of 0.95
+            test_pseudo_indices = (y_pred_proba_max >= curr_threshold)
+
+            # test_pseudo_indices = (y_pred_proba_max >= threshold)
+        else:
+            # Select a random 30% of the data to use as pseudo
+            test_pseudo_indices = pd.Series(data=False, index=y_pred_proba_og.index)
+            test_pseudo_indices_true = test_pseudo_indices.sample(frac=0.3, random_state=0)
+            test_pseudo_indices[test_pseudo_indices_true.index] = True
+
+        test_pseudo_indices = test_pseudo_indices[test_pseudo_indices]
+
+        return test_pseudo_indices
+
+    def _run_pseudo_labeling(self, hyperparameters: dict, test_data: pd.DataFrame, max_iter: int):
+        previous_score = self.info()['best_model_score_val']
+        y_pred_holdout = pd.Series()
+        X_test = test_data.copy()
+        test_data = self._learner.feature_generator.transform(test_data)
+
+        for i in range(max_iter):
+            y_pred_proba = self.predict_proba(data=X_test)
+            y_pred = self.predict(data=X_test)
+            test_pseudo_idxes_true = self._filter_pseudo(y_pred_proba_og=y_pred_proba)
+
+            if len(test_pseudo_idxes_true) < 1:
+                return self
+
+            test_pseudo_idxes = pd.Series(data=False, index=y_pred_proba.index)
+            test_pseudo_idxes[test_pseudo_idxes_true.index] = True
+
+            y_pred_holdout = y_pred_holdout.append(y_pred.loc[test_pseudo_idxes_true.index], verify_integrity=True)
+            X_pseudo = test_data.loc[y_pred_holdout.index]
+            y_pseudo = self._learner.label_cleaner.transform(y_pred_holdout)
+            self.fit_extra(hyperparameters=hyperparameters, X_pseudo=X_pseudo, y_pseudo=y_pseudo,
+                           name_suffix=PSEUDO_MODEL_SUFFIX.format(iter=(i+1)))
+            current_score = self.info()['best_model_score_val']
+
+            if previous_score >= current_score:
+                return self
+            else:
+                # Cut down X_test to not include pseudo labeled data
+                X_test = X_test.loc[test_pseudo_idxes[test_pseudo_idxes == False].index]
+                previous_score = current_score
+
+        return self
+
+    def pseudo_label_fit(self, test_data: pd.DataFrame, max_iter: int = 5, **kwargs):
```

Comment:
for future extensions to semi-supervised learning, shouldn't `test_data` be renamed to something more general like: `pseudolabeled_data`?

Reply:
yea I couldn't think of a good name that covers everything. The name should ideally cover already pseudo labeled data coming in and non-labeled data